src/algokit/core/sandbox.py

```diff
@@ -120,6 +121,49 @@ def ps(self, service_name: str | None = None) -> list[dict[str, Any]]:
         assert isinstance(data, list)
         return cast(list[dict[str, Any]], data)
 
+    def _get_local_image_version(self, image_name: str) -> str:
+        """
+        Get the local version of a Docker image
+        """
+        arg = "{{.RepoDigests}}"
+        local_version = run(
+            ["docker", "image", "inspect", image_name, "--format", arg],
+            cwd=self.directory,
+            bad_return_code_error_message="Failed to get image inspect",
+        )
+        # Remove brackets and split on '@' symbol and get the SHA hash
+        return local_version.output[1:-1].split("@")[1]
+
+    def _get_latest_image_version(self, image_name: str) -> str:
+        """
+        Get the latest version of a Docker image from Docker Hub
+        """
+        args = image_name.split(":")
+        name = args[0]
+        tag = args[1] if len(args) > 1 else "latest"
+        url = f"https://registry.hub.docker.com/v2/repositories/{name}/tags/{tag}"
+        try:
+            data = httpx.get(url=url)
+            return str(data.json()["digest"])
+        except Exception as err:
+            logger.debug(f"Error checking indexer status: {err}", exc_info=True)
+            return ""
```

Comment:
I think this should return None here if we can't fetch the latest, and then skip the warning if None is returned so that users don't get warnings if we can't determine the latest version